hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/protocolPB/OMPBHelper.java

```diff
@@ -122,10 +121,9 @@ public static OzoneAcl convertOzoneAcl(OzoneAclInfo aclInfo) {
       throw new IllegalArgumentException("ACL type is not recognized");
     }
 
-    List<IAccessAuthorizer.ACLType> aclRights = new ArrayList<>();
-    for (OzoneAclRights acl : aclInfo.getRightsList()) {
-      aclRights.add(ACLType.valueOf(acl.name()));
-    }
+    BitSet aclRights = new BitSet(aclInfo.getRightsList().size());
```

Comment:
We should not use this to set the BitSet size *aclInfo.getRightsList().size()*

Reply:
and do we really need BitSet, since we know how many bits can set apriori ? just a question

Reply:
Bitset is concise,space efficient and enforces order since we are settings them in order of our enum list. I am open to any suggestion to improve it.

Reply:
I would have used a int since you really need only 9 bits. But I come from a C background not sure if Java does that at all.